Source/DafnyRuntime/DafnyRuntimeRust/src/lib.rs

```diff
@@ -9,6 +16,346 @@ pub use num::NumCast;
 pub use num::Zero;
 pub use itertools;
 
+// An atomic box is just a RefCell in Rust
+pub type SizeT = usize;
+
+/******************************************************************************
+ * Sequences
+ ******************************************************************************/
+
+// Three elements
+// ArraySequence(var isString: bool, nodeCount: SizeT, immutable_array)
+// ConcatSequence(var isString: bool, nodeCount: SizeT, left: Sequence, right: Sequence)
+// LazySequence(length: BigInteger, box: RefCell<Rc<Sequence<T>>>)
+// We use the named version using {...}, and use snake_case format
+
+// The T must be either a *const T (allocated) OR a Reference Counting (immutable)
+
+#[allow(dead_code)]
+mod dafny_runtime_conversions {
+    use num::BigInt;
+
+    use crate::Sequence;
+    use crate::Map;
+    use std::cell::RefCell;
+    use std::collections::HashMap;
+    use std::rc::Rc;
+    use std::hash::Hash;
+    pub fn dafny_int_to_bigint(i: &Rc<BigInt>) -> BigInt {
+        i.as_ref().clone()
+    }
+    pub fn bigint_to_dafny_int(i: &BigInt) -> Rc<BigInt> {
+        Rc::new(i.clone())
+    }
+
+    pub fn dafny_sequence_to_vec<T, X>(s: &Sequence<T>, elem_converter: fn(&T) -> X) -> Vec<X>
+      where T: Clone
+    {
+        let mut array: Vec<T> = Vec::with_capacity(s.cardinality());
+        Sequence::<T>::append_recursive(&mut array, s);
+        array.iter().map(|x| elem_converter(x)).collect()
```

Comment:
Wouldn't it be more efficient (if slightly less simple) to create a `Vec<X>` and populate it with converted elements directly? That way you'd avoid the intermediate `Vec<T>`.

I don't know Rust iterators well enough yet to know the best way to implement that, but it's probably worth figuring out as a follow-up, since this conversion method is going to be used a lot on potentially large data.